### PR TITLE
Home page correction as per latest master code

### PR DESCRIPTION
**Pull request by ashish173**
This PR fixes some of the issues which were present in home page as a result of latest master code which diverged after `home_page_2` had not been updated for some time.

* Changed navbar to sticky.
* Fixed spacing issues caused due to fixed navbar.
* Removed global body colour in favour of page bg colour.
* Fixed cookie block z-index caused due to other code added in master.
* Removed some browser specific prefixes.
* Still not removed style.css as #55 is still open.

### CODE REVIEW

File: home.html [hunk 1]
```diff
@@ -734,7 +734,7 @@
           <div class="carousel-item active">
             <img src="images/carousel-img1.jpg" alt="First slide">
             <div class="carousel-caption">
-              <div class="container page-header main-header">
```
Comment (mikeselander) [inline home.html]: The `main-header` styles are almost exclusively overwritten in the and thusly cannot be trusted to work correctly. Is `_header` or `_carousel-header` supposed to be the source of truth for these?

File: src/scss/common/_static-four-column.scss [hunk 2]
```diff
@@ -25,51 +21,45 @@
 				color : $steps-deactive-text;
 				margin-bottom: 24px;
 			}
-
-			p, 
-			a {
-				color: $blue-text;
+			p {
+				color: #1a1a1a;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: No color codes should be used or defined outside of the `variables` file and this color is also not in the color palette and therefore should not be used.

File: src/scss/common/_static-four-column.scss [hunk 3]
```diff
@@ -25,51 +21,45 @@
...
-
+			
 			.blue-button { 
 				width: 100%; 
 			}
-
 			.four-coloum-information {
 				float: left;
 				min-height: 230px;
+				// margin-bottom: 60px;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Commented code

File: src/scss/common/_static-four-column.scss [hunk 4]
```diff
@@ -80,80 +70,66 @@
...
 					margin-bottom: 40px;
 				}
-				
 				p {
 					margin-bottom: 24px;
 				}
-			
 				.four-coloum-information {
 					min-height: 220px;
+					// margin-bottom: 74px;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Commented code.

File: src/scss/common/_static-four-column.scss [hunk 5]
```diff
@@ -80,80 +70,66 @@
...
-
+				text-align: left;
+	  			margin-bottom: 16px;
+	  		}
+	  		p {
+	  			margin-bottom: 24px;
+	  		}
 				.four-coloum-information {
 					min-height: 220px;
+					// margin-bottom: 86px;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Commented code.

File: src/scss/common/_static-four-column.scss [hunk 6]
```diff
@@ -168,26 +144,23 @@
...
 					margin-bottom: 32px;
 				}
-
 				p {
 					margin-bottom: 48px;
 				}
-
 				.four-coloum-information {
 					min-height: 220px;
+					// margin-bottom: 104px;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Commented code. Why are all of these commented?

File: src/scss/partials/_variables.scss [hunk 7]
```diff
@@ -104,5 +104,5 @@ $small-width: 575px;
 $medium-width: 768px;
 $medium-large-width: 992px;
 $large-width: 1024px;
-$large-large-width: 1480px;
-$extra-large-width: 1538px;
+$large-large-width: 1200px;
```
Comment (mikeselander) [inline src/scss/partials/_variables.scss]: Why are you changing all of these breakpoints? These will change across the whole site and should thusly be considered very carefully.

File: src/scss/style.scss [hunk 8]
```diff
@@ -60,7 +60,7 @@
 
 body {
   overflow-x: hidden;
-  background: $body-bg;
+  // background: $body-bg;
```
Comment (mikeselander) [inline src/scss/style.scss]: Why is this getting commented out?

As far as I know, we still need a background color on the site. We have to have a fallback of some kind that most pages will obey.

File: src/scss/partials/_variables.scss [hunk 9]
```diff
@@ -46,7 +46,7 @@ $primary-button-hover:  #ee562d;
 $secondary-button-hover:#9af9f3;
 $secondary-button:      #86eee7;
 $step-normal-bg:        #E4E7DA;
-$steps-deactive-text:   #06747A;
+$steps-deactive-text:   #0A4F45;
```
Comment (mikeselander) [inline src/scss/partials/_variables.scss]: I cannot find this on the color palette so we need to find a valid alternative.

**Comment by mikeselander** [comment]
@ashish173 FYI I merged `master` in here since we had a ton of changes so please make sure to pull down the latest before making any changes :)

**Comment by mikeselander** [comment]
Also, I forgot to mention this in my earlier review, but nice job on the smaller, more granular commits!

File: src/scss/common/_static-four-column.scss (hunk 2 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: @mikeselander I got a review doc from @viniromualdo where he asked me to change the colour of the para to this colour (`#1a1a1a`), I agree this is not in colour palette and used only once so I didn't create a variable as per your previous guidelines. Let me know how to proceed.

File: src/scss/partials/_variables.scss (hunk 9 above)
Comment (ashish173) [reply src/scss/partials/_variables.scss]: @mikeselander I got a change request from @viniromualdo to change all the colours with `#06747A` to `#0A4F45` so I proceeded with the change. Do let me know how to proceed with this one too?

File: src/scss/partials/_variables.scss (hunk 7 above)
Comment (ashish173) [reply src/scss/partials/_variables.scss]: Yes, Initially large-large-width was set to 1200, which was later changed to 1480 as per request from the designers. However, this one I forgot to revert while I had changed it to 1200 for development purpose, good catch, I'll revert it. Thanks!

File: src/scss/style.scss (hunk 8 above)
Comment (ashish173) [reply src/scss/style.scss]: With this property in place we were seeing 2 colours on from the gradient and one from this class background property. My thinking is we should be defining page level classes and then background should go into them! let me know what you think? I have done the same with home page(created a class and gradient is in that). 

About default, we should do something which should not apply if there is background is not present and not conflict with it.

File: home.html (hunk 1 above)
Comment (ashish173) [reply home.html]: Initially, I had used header as the source of truth for this but recently I realised that it was not a good idea, why? because some styles were conflicting from the header and there is as such no said rule that these heading etc in carousel header are the same as that in the header so I think its better to go with separate designs in the `_carousel-header` only rather than relying on `_header`. What do you think?

File: src/scss/common/_static-four-column.scss (hunk 3 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: removing it.

File: src/scss/common/_static-four-column.scss (hunk 4 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: removing it.

File: src/scss/common/_static-four-column.scss (hunk 5 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: removing it.

File: src/scss/common/_static-four-column.scss (hunk 6 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: I was experimenting with mobile designs and realised it was adding extra margins in large screens so wanted to remove it. I'll remove them.

File: home.html (hunk 1 above)
Comment (mikeselander) [reply home.html]: I'm fine with these being separate styles, but they need specific classes as the carousel styles are applying to and overriding the normal styles.

**Comment by mikeselander** [comment]
@viniromualdo there are several color collisions in here that were requested by you - can we get matching colors from the palette for `#1a1a1a` and `#0A4F45`?

File: src/scss/style.scss (hunk 8 above)
Comment (mikeselander) [reply src/scss/style.scss]: The page-level background classes are OK, but we still need a fallback that most pages will use that the specific classes can override. We still need _something_ on the `body` for the rest of the pages.

File: src/scss/common/_navbar.scss [hunk 10]
```diff
@@ -124,6 +123,14 @@
...
         &.donatebtn {
           width: 15%;
+          a {
+            color: $grey;
+            background: #68dfde;
+          }
+          a:hover {
+            color: $black;
+            background: #77ebe0;
+          }
```
Comment (ashish173) [inline src/scss/common/_navbar.scss]: @mikeselander we need to look at these as well for colour palettes, since you are working on colour palettes so tagging you here.

Comment (mikeselander) [reply src/scss/common/_navbar.scss]: We do, trying to avoid fixing the existing color issues for a little while since there's so much to bite off at the moment. We can simply use the new variables moving forward and fix the existing issues as they arise or after we've got the layout sorted.

File: src/scss/common/_carousel-header.scss [hunk 11]
```diff
@@ -103,20 +99,36 @@
 }
 
 // Temp fix for button placement
-@media screen and (min-width: 310px) and (max-width: 768px) {
-  .carousel-caption {
-    padding-bottom: 0;
-    .main-header {
-      position: relative;
+@media screen and 
+  (min-width: $extra-small-width) and
```
Comment (mikeselander) [inline src/scss/common/_carousel-header.scss]: Why does this have an end max-width to it? This doesn't follow mobile first which is fine only if there is a good reason for it.

File: package-lock.json [hunk 12]
```diff
@@ -108,7 +108,7 @@
         "readable-stream": {
           "version": "2.3.3",
           "resolved": "https://registry.npmjs.org/readable-stream/-/readable-stream-2.3.3.tgz",
-          "integrity": "sha512-m+qzzcn7KUxEmd1gMbchF+Y2eIUbieUaxkWtptyHywrX0rE8QEYqPC07Vuy4Wm32/xE16NcdBctb8S0Xe/5IeQ==",
```
Comment (mikeselander) [inline package-lock.json]: Every change in this file needs to be reverted. It looks like you have an out-of-date install of NPM so updating it should prevent this from happening.

File: src/scss/common/_carousel-header.scss [hunk 13]
```diff
@@ -103,20 +99,36 @@
 }
 
 // Temp fix for button placement
```
Comment (mikeselander) [inline src/scss/common/_carousel-header.scss]: Is this still temporary? Looks like it does a lot more than change a button placement.

File: src/scss/common/_carousel-header.scss [hunk 14]
```diff
@@ -103,20 +99,36 @@
...
+  (max-width: 767px) {
+  .carousel-header {
+    .carousel-caption {
       padding-bottom: 0;
-      height: 100%;
-      h3 {
-        margin-bottom: 0;
+      .main-header {
+        position: relative;
+        padding-bottom: 0;
```
Comment (mikeselander) [inline src/scss/common/_carousel-header.scss]: Do we absolutely ned all of these definitions here? Remember that this inherits everything defined in the non-breakpointed styles.

File: src/scss/common/_happy-point-block.scss [hunk 15]
```diff
@@ -145,150 +149,144 @@
...
+/*css start for subscribe-form*/
+.subscribe-form {
+  margin-left: -15px;
+  position: relative;
+}
+.subscribe-form label {
+  display: none;
+}
+.subscribe-form .en__field {
+  width: 41%;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: These additional styles are not directly related to the Happy Point block. They appear to be directly related to a specific form and should be in their own file.

File: src/scss/common/_happy-point-block.scss [hunk 16]
```diff
@@ -145,150 +149,144 @@
...
+  float: left;
+  padding: 0 15px;
+}
+.subscribe-form .en__field__input {
+  width: 100%;
+  height: 42px; 
+  border-radius: 0;
+  border:0;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: The spacing on these added rules is all over the place. Please fix.

File: src/scss/common/_happy-point-block.scss [hunk 17]
```diff
@@ -145,150 +149,144 @@
...
+  float: left;
+  padding: 0 15px;
+}
+.subscribe-form .en__submit {
+  width: 25%;
+  float: left;
+  padding: 0 15px;
+}
+.subscribe-form .en__submit button {
+  background-color: #F36D3A;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: @ashish173 we've talked about not adding more color codes - why is there a color code here?

File: src/scss/common/_happy-point-block.scss [hunk 18]
```diff
@@ -145,150 +149,144 @@
...
+  height: 42px;
+  line-height: 42px;
+  color: #fff;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: Color code.

File: src/scss/common/_happy-point-block.scss [hunk 19]
```diff
@@ -145,150 +149,144 @@
...
+  text-transform: uppercase;
+  width: 100%;
+  font-size: 16px;
+  border: 1px solid transparent;
+  padding: 0 15px;
+  border-radius: 5px;
+}
+.subscribe-form .en__submit button:hover {
+  background-color: #e02d00;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: Color code.

File: src/scss/common/_happy-point-block.scss [hunk 20]
```diff
@@ -145,150 +149,144 @@
...
-        margin-bottom: 10px;
-        float: left;
-      }
-      .subscribe-text {
-        float: right;
-        width: 24%;
-        padding-right: 15px;
-      }
-    }
+@media (min-width: 768px) {
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: Why are there breakpoints that don't use our breakpoint mixins?

File: src/scss/common/_happy-point-block.scss [hunk 21]
```diff
@@ -145,150 +149,144 @@
...
+  .subscribe-form .en__field {
+    width: 50%;
+    margin-bottom: 16px;
+  }
+  .subscribe-form .en__submit {
+    width: 100%;
+    margin-bottom: 10px;
+  }
+  .subscribe-text {
+    float: right;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: Why is this floating if it's 100%?

File: src/scss/common/_happy-point-block.scss [hunk 22]
```diff
@@ -145,150 +149,144 @@
...
-        float: right;
-        width: 24%;
-        padding-right: 15px;
-      }
-    }
\ No newline at end of file
+@media (min-width: 992px) {
+  .subscribe-form .en__field {
+    width: 41%;
+    margin-bottom: 16px;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: These breakpoints are not using the previously-inherited styles. Only define what absolutely has to be over-ridden. Almost none of these styles are necessary.

File: src/scss/common/_navbar.scss (hunk 10 above)
Comment (mikeselander) [reply src/scss/common/_navbar.scss]: Another note here: should these styles _only_ override on the home page or should they apply on every single page?

File: src/scss/common/_navbar.scss [hunk 23]
```diff
@@ -594,3 +473,19 @@
...
+        img {
+          height: 20px;
+        }
+      }
+      .mobile-search-wrap {
+        padding: 5px 5px 0 0;
+      }
+    }
+  }
+}
```
Comment (mikeselander) [inline src/scss/common/_navbar.scss]: Missing EOF line

File: src/scss/common/_static-four-column.scss [hunk 24]
```diff
@@ -1,195 +1,172 @@
...
+    font-size: rem(30);
+    color: $steps-deactive-text;
+    line-height: 36px;
+    text-align: center;
+    margin: 60px 0 35px 0;
+  }
+  .four-coloum-wrap {
+    margin-bottom: 35px;
+    .four-coloum-symbol {
+      text-align: center;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Why is so much of this file being overridden?

Will this affect other areas of the site?

@alexcavender since you last worked on this section can you take a look at these changes?

File: src/scss/common/_static-four-column.scss [hunk 25]
```diff
@@ -1,195 +1,172 @@
...
+      margin-bottom: 16px;
+    }
+    .four-coloum-info {
+      h5 {
+        text-align: center;
+        font-size : 22px;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Not using `rem()` here

File: src/scss/common/_static-four-column.scss [hunk 26]
```diff
@@ -1,195 +1,172 @@
...
+        line-height : normal;
+        font-weight: 500;
+        color : $steps-deactive-text;
+        margin-bottom: 24px;
+      }
+      p {
+        color: #1a1a1a;
```
Comment (mikeselander) [inline src/scss/common/_static-four-column.scss]: Color code

File: src/scss/modules/_buttons.scss [hunk 27]
```diff
@@ -257,7 +257,7 @@ Stylesheet: Button Styles
     }
 
     &.btn-primary, &.yellow-button {
-      padding: 16px 10px;
+      // padding: 16px 10px;
```
Comment (mikeselander) [inline src/scss/modules/_buttons.scss]: Why are you commenting code out??

The code is either deleted or it stays - no commenting.

File: src/scss/style.scss (hunk 8 above)
Comment (mikeselander) [reply src/scss/style.scss]: Either way in this situation - this *should not be commented out* if it needed to be deleted. We cannot be commenting code instead of deleting it in the codebase. If it goes it really goes, if it stays it's doing something.

File: src/scss/common/_happy-point-block.scss [hunk 28]
```diff
@@ -145,150 +149,144 @@
...
+  border-radius: 0;
+  border:0;
+  padding: 10px;
+}
+.subscribe-form .en__field--select {
+  width: 33%;
+  float: left;
+  padding: 0 15px;
+}
+.subscribe-form .en__submit {
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: The `en` styles appear to be directly related to a specific language. What happens when a form in a different language is loaded?

File: src/scss/common/_happy-point-block.scss [hunk 29]
```diff
@@ -145,150 +149,144 @@
...
+  width: 25%;
+  float: left;
+  padding: 0 15px;
+}
+.subscribe-form .en__submit button {
+  background-color: #F36D3A;
+  height: 42px;
+  line-height: 42px;
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: Line heights should not use units.

File: src/scss/common/_happy-point-block.scss [hunk 30]
```diff
@@ -145,150 +149,144 @@
...
+  .subscribe-form .en__submit {
+    width: 25%;
+    margin-bottom: 10px;
+  }
+  .subscribe-text {
+    float: right;
+    width: 24%;
+    padding-right: 15px;
+  }
+}
```
Comment (mikeselander) [inline src/scss/common/_happy-point-block.scss]: Missing EOF line

File: src/scss/common/_navbar.scss [hunk 31]
```diff
@@ -594,3 +473,19 @@
     }
   }
 }
+
+@media screen and (max-width: 375px) {
+  .navbar {
+    .mobile-header-wrap {
+      padding: 9px 0;
+      a {
+        img {
```
Comment (mikeselander) [inline src/scss/common/_navbar.scss]: Why is this so deeply nested?

**Comment by mikeselander** [comment]
@ashish173 additionally this keeps getting larger and larger in size - it needs to be capped off and not just a general catch-all for any home page issues. This has the potential to cause too many regressions.

File: src/scss/common/_happy-point-block.scss (hunk 17 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander Please note that this part of the file is plain CSS as it has to be moved to a different server. so this is sort of an exception to that rule.

File: src/scss/common/_happy-point-block.scss (hunk 18 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander Please note that this part of the file is plain CSS as it has to be moved to a different server. so this is sort of an exception to that rule.

File: src/scss/common/_happy-point-block.scss (hunk 19 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander Please note that this part of the file is plain CSS as it has to be moved to a different server. so this is sort of an exception to that rule.

File: src/scss/common/_happy-point-block.scss (hunk 20 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander Please note that this part of the file is plain CSS as it has to be moved to a different server. so this is sort of an exception to that rule.

File: src/scss/style.scss (hunk 8 above)
Comment (ashish173) [reply src/scss/style.scss]: I'll be removing this, for now, we can work on the background strategy in a separate PR creating an issue to track it #78

File: src/scss/common/_navbar.scss (hunk 10 above)
Comment (ashish173) [reply src/scss/common/_navbar.scss]: This should be the donate button style for all pages where there is donate button in the navbar. BTW I added the vars for the colour codes.

File: package-lock.json (hunk 12 above)
Comment (ashish173) [reply package-lock.json]: Okay, I'll do that!

File: src/scss/common/_carousel-header.scss (hunk 11 above)
Comment (ashish173) [reply src/scss/common/_carousel-header.scss]: The styling that I have applied in this I didn't want it to be inherited to the larger resolution screens as I didn't want to override all those in the rest of the media queries. So it felt right to use that.

File: src/scss/common/_carousel-header.scss (hunk 14 above)
Comment (ashish173) [reply src/scss/common/_carousel-header.scss]: @mikeselander don't seem to be repeating the definitions around the block you commented, let me know if I am missing something.

File: src/scss/common/_happy-point-block.scss (hunk 15 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander Please note this form lives on the separate server and the styling for it also goes on that server. We use this as an IFRAME on our webpage. I had added it here so that @sagarsdeshmukh could simply copy and paste into the server apart from that!

My initial thinking was that since this form would never ever appear with any other block so having it in `_happy-point.scss`, I don't see much of a problem!

File: src/scss/common/_happy-point-block.scss (hunk 28 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander I was told that It is safe to assume the markup of this form would not change and I was instructed to use the classes present there so I think we are good with this.

File: src/scss/common/_happy-point-block.scss (hunk 29 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: Fixed this.

File: src/scss/common/_happy-point-block.scss (hunk 22 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander Made changes as per your suggestion, please review.

File: src/scss/common/_happy-point-block.scss (hunk 30 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: fixed it.

File: src/scss/common/_navbar.scss (hunk 31 above)
Comment (ashish173) [reply src/scss/common/_navbar.scss]: removed extra deep nesting.

File: src/scss/common/_navbar.scss (hunk 23 above)
Comment (ashish173) [reply src/scss/common/_navbar.scss]: added one.

File: src/scss/common/_static-four-column.scss (hunk 25 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: changed it to use rem().

File: src/scss/common/_static-four-column.scss (hunk 26 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: added the variable for colour code.

File: src/scss/modules/_buttons.scss (hunk 27 above)
Comment (ashish173) [reply src/scss/modules/_buttons.scss]: removed the commented code.

**Comment by ashish173** [comment]
@mikeselander making changes only on your feedback now not adding major fixes and changes so let's get it to merge once you think its good for merge.

File: src/scss/common/_static-four-column.scss (hunk 24 above)
Comment (ashish173) [reply src/scss/common/_static-four-column.scss]: @alexcavender do let me know what you think about these changes.

File: src/scss/common/_happy-point-block.scss (hunk 21 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: fixed it.

File: src/scss/common/_happy-point-block.scss (hunk 20 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: @mikeselander refer my comments above! this goes to another server as is. so had to be pure css.

File: src/scss/common/_carousel-header.scss (hunk 13 above)
Comment (ashish173) [reply src/scss/common/_carousel-header.scss]: Nopes, over time the behaviour changed, removed the comment!

File: src/scss/common/_happy-point-block.scss (hunk 16 above)
Comment (ashish173) [reply src/scss/common/_happy-point-block.scss]: fixed it!

File: home.html (hunk 1 above)
Comment (ashish173) [reply home.html]: Let's discuss today for this one, I'll be available for chat on rocket.chat.

**Comment by mikeselander** [comment]
Merging this PR so that we can move forward on smaller sets of changes and adjustments and so that we can move forward with confidence that master aligns with `gh-pages`.
